internal/host/host.go

```diff
@@ -320,19 +339,28 @@ func (m *Manager) RefreshStatus(ctx context.Context, h *models.Host, db *gorm.DB
 	if err != nil {
 		return err
 	}
-	conditions, validationsResults, err := m.rp.preprocess(vc)
+	conditions, newValidationRes, err := m.rp.preprocess(vc)
 	if err != nil {
 		return err
 	}
-	if err = m.reportValidationStatusChanged(ctx, vc, h, validationsResults); err != nil {
+	currentValidationRes, validationsChanged, err := m.checkValidationChanged(ctx, h, newValidationRes)
+	if err != nil {
 		return err
 	}
+	if validationsChanged {
+		// reporting validations change in status needs to compare the new validations results
```

Comment:
I would suggest "Validation status changes are detected when new validations are different from the current validations in the DB. For changes to be detected and reported correctly, the comparison needs to be performed before the new validations are updated to the DB".

Reply:
No problem, updated, thanks for the review.